### ISSUE TITLE
ZipFile::saveCollectionToArchive creates invalid zip file (bad CRC)

### ISSUE THREAD

**Issue by mzf-guest** [Zipios/Zipios#38]
Hello,

I'm using ```ZipFile::saveCollectionToArchive``` to create a zip file from a ```DirectoryCollection``` but the resulting zip file seems corrupted.

To reproduce:

- have a "test" folder with some files inside (test.dat file in my case)
- run the following code which should create a test.zip archive from your "test" folder: 
```cpp
    zipios::DirectoryCollection directoryCollection("test");
    std::ofstream tempZipStream("test.zip", std::ios_base::binary | std::ios::out);
    zipios::ZipFile::saveCollectionToArchive(tempZipStream, directoryCollection);
    tempZipStream.close();
```

- unzip the resulting file via the command line:
```
$ unzip test.zip 
Archive:  test.zip
   creating: test/
 extracting: test/test.dat            bad CRC b62377d5  (should be 00000000)
```

Other extraction softwares (like archive managers) also failed to extract files in the zip file.

I'm using zipios 2.2.6 on Linux 64bits (Debian).

Thanks,

**Comment by AlexisWilke** [comment]
Okay, I can reproduce the issue. Same error, it finds a CRC but wants 0 instead.

**Comment by AlexisWilke** [comment]
Sorry it took me so long, but well... things got in the way.

I found the issue, when adding a small file, the system switch to "STORED" instead of "COMPRESSED". That means we hit a different case and in that specific case, the library was not calculating the CRC32 at all. Adding a call to the zlib fixed the issue in the streaming scheme.

I also enhanced your test and added three files instead of just one. The empty file created yet another issue in your version of the fix. I kept the `computerCRC32()`, but it's not used to calculate the missing CRC32, it will just be a helper function one can use to get the CRC32 of one of their file. It could be useful in the tests but I did not look into fixing that yet.

Thank you for reporting this issue. Really bad one!
